lib/guru-utils.js

```diff
@@ -0,0 +1,82 @@
+'use babel'
+
+import path from 'path'
+import {getEditor, getCursorPosition, isValidEditor, utf8OffsetForBufferPosition} from './utils'
+
+const scopedModes = ['callees', 'callers', 'callstack', 'pointsto', 'whicherrs', 'peers']
+
+function buildGuruArchive (editor = null) {
+  let archive = ''
+  const editors = editor ? [editor] : atom.workspace.getTextEditors()
+  for (const e of editors) {
+    if (e.isModified() && isValidEditor(e)) {
+      archive += e.getPath() + '\n'
+      archive += Buffer.byteLength(e.getText(), 'utf8') + '\n'
+      archive += e.getText()
+    }
+  }
+  return archive
+}
+
+function computeArgs (mode, editor = getEditor(), pos = currentCursorOffset(editor)) {
+  if (!mode || !editor || !pos) {
+    return
+  }
+
+  const filePath = editor.getPath()
+  const args = ['-json']
+  if (scopedModes.includes(mode)) {
+    let relPath = atom.project.relativizePath(filePath)
+    if (relPath && relPath.length > 0 && relPath[0] !== null) {
+      // Make it relative to GOPATH because guru tool requires it.
+      if (this.goconfig && this.goconfig.locator) {
```

Comment:
`computeArgs` is a global function and it never receives a proper value for `this`. So this will never work.